### PR TITLE
Use Directory.Build.props instead of tools\common.props.

### PR DESCRIPTION
**Pull request by eerhardt**
This ensures new projects get the global settings by default.

### CODE REVIEW

File: tests/Microsoft.Data.DataFrame.Tests/DataFrameTests.cs [hunk 1]
```diff
@@ -1020,9 +1020,9 @@ public void TestGroupBy()
         [Fact]
         public void TestGoupByDifferentColumnTypes()
         {
-            void GroupCountAndAssert(DataFrame df)
+            void GroupCountAndAssert(DataFrame frame)
```
Comment (ahsonkhan) [inline tests/Microsoft.Data.DataFrame.Tests/DataFrameTests.cs]: Lol. Unrelated parameter renames got snuck into this PR :)

Comment (eerhardt) [reply tests/Microsoft.Data.DataFrame.Tests/DataFrameTests.cs]: These changes were related. Before, this project didn’t have warnings as errors turned on. With directory build props, now it is on for all projects. These places were causing errors because the names conflicted with variables in the outer scope.

Comment (ahsonkhan) [reply tests/Microsoft.Data.DataFrame.Tests/DataFrameTests.cs]: Ah! I didn't notice that. `string data` used twice within the same method in different scopes. Nice. Thanks for clarifying.
